src/DocumentType/AbstractDocument.php

```diff
@@ -68,6 +73,10 @@ public function getListingClass(): string
             return $this->getSubType() . '\Listing';
         }
 
+        if (in_array($this->getType(), [DocumentInterface::TYPE_ASSET], true)) {
+            return $this->getSubType() . '\Listing';
```

Comment:
Unfortunately this does not work when implementing `getSubType` for assets analogous to the implementation for documents.

Consider the following:

```php
<?php

namespace App\Elasticsearch\Document;

use Pimcore\Model\Asset\Image;
use Valantic\ElasticaBridgeBundle\DocumentType\AbstractDocument;
use Valantic\ElasticaBridgeBundle\DocumentType\DocumentInterface;

class ImageDocument extends AbstractDocument
{
    public function getType(): string
    {
        return DocumentInterface::TYPE_ASSET;
    }

    public function getSubType(): string
    {
        return Image::class;
    }
}
```

However, `Pimcore\Model\Asset\Image\Listing` does not exist. Hence something similar to the condition below is also needed for assets.